interface/HemisphereMixer.h

```diff
@@ -268,9 +268,9 @@ template <class EventClass> class HemisphereMixer : public BaseOperator<EventCla
         }
 
         std::vector<std::size_t> index_nns(knn_);
-        std::vector<double> dist_nns(knn_);
+        //std::vector<double> dist_nns(knn_);
         // query kdtree
-        index_m_.at(h_cat)->knnSearch(&h_vars[0], knn_, &index_nns[0], &dist_nns[0]);
+        index_m_.at(h_cat)->knnSearch(&h_vars[0], knn_, &index_nns[0], &ev.best_match_distances_);
```

Comment:
this has to be &ev.best_match_distances_[0] so it is a pointer of type double*, but the issue the way your are doing it now is that the ev.best_match_distances_ will be rewritten for every nth closest hemisphere , in addition, you need to consider that there are two hemispheres per event, that is why I suggested a vector of vectors of double initially.